test/types/tuple/errors/tuple-init.bad

```diff
@@ -0,0 +1,8 @@
+$CHPL_HOME/modules/internal/ChapelBase.chpl:1347: internal error: UTI-MIS-0939 chpl version 1.27.0 pre-release (b832268935)
```

Comment:
There's three parts of this that may change over time without the future being resolved that could cause problems for us.  Thankfully, start_test is set up so that it will allow variation on those parts, but I think you need to adjust so it knows to allow the variation.

I think the line should look like this instead:
```
$CHPL_HOME/modules/internal/ChapelBase.chpl:nnnn: internal error: UTI-MIS-nnnn chpl version mmmm
Where the parts that differ are         here^                         here^         and here^
```

Reply:
Hmm.  I based that .bad file on other internal error ones in the test/types/tuple tree, which also have concrete values for those fields, so I thought start_test's allowances would cover that.

And testing now, I still get a clean match when I change all three of those fields in the .bad file.  Whereas if I make it expect an `infernal error` it reports a mismatch.  (So it's not that I'm seeing the good match only because I made those changes to the wrong file or something.)

Reply:
The test does pass today because those line numbers haven't changed.  The true check is if adding a blank line to ChapelBase.chpl before line 1347 results in a failure for the .future

Reply:
I mean, I made this change

```
-$CHPL_HOME/modules/internal/ChapelBase.chpl:1347: internal error: UTI-MIS-0939 chpl version 1.27.0 pre-release (b832268935)
+$CHPL_HOME/modules/internal/ChapelBase.chpl:1: internal error: UTI-MIS-2468 chpl version 3.14.1 pre-release (abc123)
```

(effectively changing all four variable parts, as if the bad file had been generated by a different everything) and it stilll resulted in a clean match.  I think this indicates that using the currently-submitted .bad file with future versions of the compiler, modules, and util/misc.cpp will continue to work.

Whereas with this change

```
-$CHPL_HOME/modules/internal/ChapelBase.chpl:1347: internal error: UTI-MIS-0939 chpl version 1.27.0 pre-release (b832268935)
+$CHPL_HOME/modules/internal/ChapelBase.chpl:1347: infernal error: UTI-MIS-0939 chpl version 1.27.0 pre-release (b832268935)
```

it does report "error matching" against the bad, and says

```
< $CHPL_HOME/modules/internal/ChapelBase.chpl:nnnn: infernal error: UTI-MIS-0939 chpl version mmmm
---
> $CHPL_HOME/modules/internal/ChapelBase.chpl:nnnn: internal error: UTI-MIS-nnnn chpl version mmmm
```

so it's already nnnned and mmmmed the module line number and chpl version before comparing.  (Odd that it hasn't done the same thing to the `<` compilerError line number; has done it to the `>` one, and still reports a clean match normally; maybe it handles that one separately.)


I also added lines to ChapelBase.chpl and util/misc.cpp and built the compiler with gitid 21acc7b8a6 and got different line numbers in the error message, and a clean match against the submitted .bad file.

Reply:
Okay, that satisfies me.  I'll just merge the PR then, thanks!

Reply:
Ah, it's all done by `util/test/diff-ignoring-module-line-numbers`.  (Which by comments and callsite is applied only to the comparison of output to .bad, not the comparison to .good.  I guess there should never be an internal error in a .good, so that makes sense.)